webhook_redelivery.py

```diff
@@ -0,0 +1,412 @@
+
+    Attributes:
+        github_org: Github organisation where the webhook is registered.
+        github_repo: Github repository, where the webhook is registered. Only applicable for
+            repository webhooks.
+        id: The identifier of the webhook.
+    """
+
+    github_org: str
+    github_repo: str | None
+    id: int
+
+
+@dataclass
+class _WebhookDeliveryAttempts:
+    """The details of a webhook delivery attempt.
+
+    Attributes:
+        id: The identifier of the delivery.
+        status: The status of the delivery.
+        delivered_at: The time the delivery was made.
+        action: The action type of the delivery.
+         See https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_job for
+         possible values for the workflow_job event. May be null for other events (e.g. ping).
+        event: The event type of the delivery.
+    """
+
+    id: int
+    status: str
+    delivered_at: datetime
+    action: str | None
+    event: str
+
+
+class RedeliveryError(Exception):
+    """Raised when an error occurs during redelivery."""
+
+
+class ArgParseError(Exception):
+    """Raised when an error occurs during argument parsing."""
+
+
+def main() -> None:  # pragma: no cover this is checked by integration tests
+    """Run the module as script."""
+    args = _arg_parsing()
+
+    redelivery_count = _redeliver_failed_webhook_delivery_attempts(
+        github_auth=args.github_auth_details,
+        webhook_address=args.webhook_address,
+        since_seconds=args.since,
+    )
+
+    print(_create_json_output(redelivery_count))
+
+
+def _arg_parsing() -> _ParsedArgs:  # pragma: no cover this is checked by integration tests
+    """Parse the command line arguments.
+
+    Raises:
+        ArgParseError: If the arguments are invalid.
+
+    Returns:
+        The parsed arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description=f"{__doc__}. The script returns the amount of redelivered webhooks in JSON"
+        "format.The script assumes github app auth details to be given via env variables"
+        " via stdin. The format has to be a json object with either the only key"
+        " 'token' or the keys 'app_id', 'installation_id' and 'private_key'.,"
+        " depending on the authentication method (github token vs github app auth) used."
+    )
+    parser.add_argument(
+        "--since",
+        type=int,
+        help="The amount of seconds to look back for failed deliveries.",
+        required=True,
+    )
+    parser.add_argument(
+        "--github-path",
+        type=str,
+        help=(
+            "The path of the organisation or repository where the webhooks are registered. Should"
+            "be in the format of <organisation> or <organisation>/<repository>."
+        ),
+        required=True,
+    )
+    parser.add_argument(
+        "--webhook-id", type=int, help="The identifier of the webhook.", required=True
+    )
```

Comment:
Is this to redeliver a specific webhook?

Reply:
This is to identify a specific webhook (not the same as "webhook delivery attempt") for which delivery attempts are retrieved.

A user can register multiple webhooks on a repository/organisation and we need to identify the specific one.

I'll update the help text.